### PR TITLE
feat: upload cv option added

### PR DESCRIPTION
**Pull request by sabbirrifat**

### CODE REVIEW

File: app/talents/my-profile/page.tsx [hunk 1]
```diff
@@ -68,39 +78,76 @@ export default function MyProfile() {
...
+        type,
+        data: arrayBuffer,
+      });
+    };
+  };
+
+  const uploadeFileToBucket = async (file: FileData | null) => {
+    if (!file) return null;
+    try {
+      const postImageResponse = await fetch("/api/uploadFile", {
```
Comment (n1c01a5) [inline app/talents/my-profile/page.tsx:111]: Use kebab-case for url.

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: app/api/uploadFile/route.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { S3, PutObjectCommand, GetObjectCommand } from "@aws-sdk/client-s3";
```
Comment (n1c01a5) [inline app/api/uploadFile/route.ts:1]: Use kebab-case for url like `upload-file`.

File: app/api/uploadFile/utils.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { v4 as uuidv4 } from "uuid";
```
Comment (n1c01a5) [inline app/api/uploadFile/utils.ts:1]: Use kebab-case for file name.

File: app/api/uploadFile/utils.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+export const imageTypes = ["image/jpeg", "image/jpg", "image/png"];
```
Comment (n1c01a5) [inline app/api/uploadFile/utils.ts:3]: Purpose: move to env.

File: app/api/uploadFile/utils.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+export const generateKeyName = (fileType: string) => {
```
Comment (n1c01a5) [inline app/api/uploadFile/utils.ts:5]: The name function is not so explicit.

Maybe make explicit it's about file extension like `generateFileKeyName`.

File: app/api/uploadFile/utils.ts (hunk 3 above)
Comment (n1c01a5) [reply app/api/uploadFile/utils.ts:1]: It looks better to put this `./utils/*`.

File: app/talents/my-profile/page.tsx [hunk 6]
```diff
@@ -68,39 +78,76 @@ export default function MyProfile() {
...
-      fetchImage();
+    if (cvFile.size > 1024 * 1024 * Number(resumeUploadSizeLimit)) {
+      toast.error(`File size should be less than ${resumeUploadSizeLimit} MB`);
+      return;
     }
-  }, [file]);
+    const reader = new FileReader();
+    reader.readAsDataURL(cvFile);
+
+    reader.onload = function (event: ProgressEvent<FileReader>) {
```
Comment (n1c01a5) [inline app/talents/my-profile/page.tsx:94]: Make something more consistant. Use `e` or `event`.

File: app/talents/my-profile/page.tsx [hunk 7]
```diff
@@ -68,39 +78,76 @@ export default function MyProfile() {
...
+      // event.target.result contains the ArrayBuffer representation of the file
+      if (!event.target) return;
+      const arrayBuffer = event.target.result;
+      // Do something with the arrayBuffer, for example, send it to the server or process it further
+      console.log('ArrayBuffer:', arrayBuffer);
```
Comment (n1c01a5) [inline app/talents/my-profile/page.tsx:99]: Remove `log`.

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: app/talents/my-profile/page.tsx (hunk 7 above)
Comment (sabbirrifat) [reply app/talents/my-profile/page.tsx:99]: resolved!

File: app/talents/my-profile/page.tsx (hunk 6 above)
Comment (sabbirrifat) [reply app/talents/my-profile/page.tsx:94]: resolved!

File: app/api/uploadFile/utils.ts (hunk 5 above)
Comment (sabbirrifat) [reply app/api/uploadFile/utils.ts:5]: resolved!

File: app/api/uploadFile/utils.ts (hunk 3 above)
Comment (sabbirrifat) [reply app/api/uploadFile/utils.ts:1]: resolved!

File: app/api/uploadFile/route.ts (hunk 2 above)
Comment (sabbirrifat) [reply app/api/uploadFile/route.ts:1]: resolved!

**Review by n1c01a5 — APPROVED** [review]

**Pull request merged**